pkg/handlers/aws/mutation/ami/inject_control_plane_test.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2023 D2iQ, Inc. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package ami
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+	runtimehooksv1 "sigs.k8s.io/cluster-api/exp/runtime/hooks/api/v1alpha1"
+
+	"github.com/d2iq-labs/cluster-api-runtime-extensions-nutanix/api/v1alpha1"
+	"github.com/d2iq-labs/cluster-api-runtime-extensions-nutanix/common/pkg/capi/clustertopology/handlers/mutation"
+	"github.com/d2iq-labs/cluster-api-runtime-extensions-nutanix/common/pkg/testutils/capitest"
+	"github.com/d2iq-labs/cluster-api-runtime-extensions-nutanix/common/pkg/testutils/capitest/request"
+	"github.com/d2iq-labs/cluster-api-runtime-extensions-nutanix/pkg/handlers/generic/clusterconfig"
+)
+
+var _ = Describe("Generate AMI patches for ControlPlane", func() {
+	patchGenerator := func() mutation.GeneratePatches {
+		return mutation.NewMetaGeneratePatchesHandler("", NewControlPlanePatch()).(mutation.GeneratePatches)
+	}
+
+	testDefs := []capitest.PatchTestDef{
+		{
+			Name: "AMI set for control plane",
+			Vars: []runtimehooksv1.Variable{
+				capitest.VariableWithValue(
+					clusterconfig.MetaVariableName,
+					v1alpha1.AMISpec{ID: "ami-controlplane"},
+					clusterconfig.MetaControlPlaneConfigName,
+					v1alpha1.AWSVariableName,
+					VariableName,
```

Comment:
Maybe in a different PR we can move this to a var. Its also a little strange that `capitest.VariableWithValue` takes in a var and `variablePath`, maybe once all the tests are moved we'll be able to just take in a single var.

Reply:
Great suggestion. we have a AWSAMIControlplane var and AWSAMIWorker var or function that we can use. that way we less like to make error every time we create variable. Lets have a sample example in separate PR, once we close that, I can make changes to all other variables.